### PR TITLE
(SERVER-2465) Allow certificate chains to be in any order

### PR DESCRIPTION
**Pull request by pcarlisle**

### CODE REVIEW

**Comment by Magisus** [comment]
This makes me realize I haven't really written Java since before streams were a thing. Pretty sweet.

**Comment by pcarlisle** [comment]
yeah i wrote a ten line version of this first and then modernized it

**Comment by justinstoller** [comment]
Are we going to handle CRLs in this PR too?

**Comment by pcarlisle** [comment]
hmm i might have to break the api for that one

File: src/clojure/puppetlabs/ssl_utils/core.clj [hunk 1]
```diff
@@ -561,12 +561,17 @@
...
 (schema/defn ^:always-validate pem->ca-crl :- X509CRL
-  "Given a CRL chain, extract the first CRL"
-  [crl-chain :- Readerable]
-  (let [[crl rest-of-chain] (pem->crls crl-chain)]
+  "Given a CRL chain and CA certificate, extract the CRL issued by the
+  certificate"
+  [crl-chain :- Readerable
+   ca-cert :- X509Certificate]
+  (let [match-ca-cert? (fn [crl] (= (.getIssuerX500Principal crl)
+                                    (.getSubjectX500Principal ca-cert)))
```
Comment (justinstoller) [inline src/clojure/puppetlabs/ssl_utils/core.clj]: there any value in saving this to variable instead of calling getSubjectX500Principal every time?

**Pull request merged**